elixir/apps/api/lib/api/gateway/channel.ex

```diff
@@ -347,4 +335,30 @@ defmodule API.Gateway.Channel do
       {:reply, :ok, socket}
     end
   end
+
+  defp encode_ref(socket, channel_pid, socket_ref, resource_id, opentelemetry_headers) do
+    ref =
+      {channel_pid, socket_ref, resource_id, opentelemetry_headers}
+      |> :erlang.term_to_binary()
+      |> Base.url_encode64()
+
+    key_base = socket.endpoint.config(:secret_key_base)
+    Plug.Crypto.sign(key_base, "gateway_reply_ref", ref)
```

Comment:
@bmanifold because you can build a `ref` that would be valid and if you can guess the process ID that will make the gateway process send a message to an arbitrary client process (and maybe crash it), so there is a chance for DDoS, very hard to pull off though.